webapp/channels/src/reducers/views/rhs.ts

```diff
@@ -380,6 +380,21 @@ function editChannelMembers(state = false, action: AnyAction) {
     }
 }
 
+function shouldFocusRHS(state = false, action: AnyAction) {
+    switch (action.type) {
+    case ActionTypes.SELECT_POST:
+        return Boolean(action.postId);
+    case Threads.CHANGED_SELECTED_THREAD:
+        return Boolean(action.data.thread_id);
+    case ActionTypes.HIGHLIGHT_REPLY:
```

Comment:
This is needed when we jump to a thread from a permalink or search.

Right now, when we focus the textbox, the scroll goes to the bottom (so the textbox is visible). Hopefully when we detach the textbox from the scroll list, we can improve all this behavior.